dask/bytes/local.py

```diff
@@ -87,10 +116,25 @@ def open_files(path):
             for _path in filepaths]
 
 
+def write_files(paths, mode='wb'):
+    """ Open many files for writing.  Return delayed objects.
+
+    See Also
+    --------
+    dask.bytes.core.write_files: User function
+    """
+    myopen = delayed(open)
+    files = [myopen(_path, mode=mode,
+                   dask_key_name='write-%s' % tokenize(_path, time.time()))
+            for _path in paths]
+    return files
```

Comment:
Write, so the original intent of open_files was to return an Iterator of lines of text (or bytes).  It turned out that file objects satisfy this interface, so we just returned them, but all uses of this interface would generally just iterate over them as iterators of text.  

Originally I was thinking that write_files would perform the reverse and take a list of delayed values, each of which is a sequence of bytes, and would write that sequence of bytes to a file, returning None.  This suffices to implement `dask.bag.to_textfiles`, the function for which `write_bytes` won't necessarily work because the sequences can be larger-than-ram.  

However, what's here is also interesting, as evidenced by your solution to `DataFrame.to_csv`.

My gut feeling is that fewer issues will arise if we use `write_bytes` for `to_csv` but I really don't know.  Your solutions creating nice, feature complete File objects S3/HDFS provide strong support for the solution that you've taken here.  

I don't have a strong understanding of what the right approach is here.